### DISCUSSION TITLE
GdUnit4 TicTacToe tutorial video

### DISCUSSION THREAD

**Discussion by ACG8** [MikeSchulze/gdUnit4#156] [Show and tell]
Just posted my first gdUnit4 tutorial video (link to part 1 of a 3-part series using TicTacToe as an example):

https://www.youtube.com/watch?v=JHMTCXECS-8&t=0s

Unfortunately, at the time that I recorded it, there were a couple of issues (in particular, the indentation issue from #150 and the lack of a gdUnit4 documentation link from #153) that have since been fixed (I put a note in the video description). But hopefully, it should still be a good concrete example. I might post some more video tutorials later too.

**Comment by MikeSchulze** [comment]
hi @ACG8 

Thank you, this is a very good beginner video and shows very well how to create and run tests with GdUnit.
I am very grateful to you for this, unfortunately I currently have no time to create tutorials myself.
The more I would be happy if you create more videos and show the power of GdUnit4 ;)

I know it is a lot of work, and the bugs at the time of recording are annoying. If it is possible for you it would be nice to update the video based on version 4.1.0.
With v4.1.0 there is now also shortcut support and the documentation is in work. (the link in the inspector points to the new GdUnit4 Documentation) 

Thanks


Update:
I see that there are already three parts ;)
I would like to link your tutorials in the documentation if you allow me to do so

You can test your TicTacToe scene also by using the provided scenerunner (integreation testing)
https://mikeschulze.github.io/gdUnit4/advanced_testing/sceneRunner/

**Comment by ACG8** [comment]
Yes, I would be more than happy to have my tutorials linked in the documentation, thanks! I'm not sure about re-recording tictactoe (annoying as the bugs are), though I might add some annotations to the video so that people know the bugs are fixed, but I was also thinking of recreating other simple games (snake, tetris, etc.) to show how gdunit4 can be used, and to introduce new concepts (such as scene_runner) in each one.

**Comment by MikeSchulze** [comment]
Hi nice, 

before I link this tutorial, it would be good to update part 1 and replace the two places that refer to the bugs.
I guess you have a good video cut tool to simply replace these parts.

I would be very grateful.
I am looking forward to your next tutorials

**Comment by ACG8** [comment]
All right, I have edited part 1 to remove references to the bugs. Here is the new link (youtube doesn't allow you to replace videos, so I had to delete the old version and upload a brand new video):

https://www.youtube.com/watch?v=MW89LbaBLlw

**Comment by MikeSchulze** [comment]
Thanks for updating the part1
The GdUnit4 is now updated to v4.1.0 on AssetLib and should be installable.

**Comment by MikeSchulze** [comment]
Update!

Your tutorial is now officially linked in the [Documentation](https://mikeschulze.github.io/gdUnit4/tutorials/tdd/).

Thank you for your support.

**Reply by ACG8** [reply]
Thanks! I've also uploaded the project to github here:

https://github.com/ACG8/Tic-Tac-Toe

**Reply by MikeSchulze** [reply]
very nice I will also link it
